launch/dynamo-run/src/input/common.rs

```diff
@@ -46,22 +46,29 @@ pub async fn prepare_engine(
                 .component(endpoint_id.component.clone())?
                 .endpoint(endpoint_id.name.clone());
 
-            let mut client = endpoint.client::<NvCreateChatCompletionRequest, Annotated<NvCreateChatCompletionStreamResponse>>().await?;
-
-            match &flags.router_mode {
+            let client = endpoint.client().await?;
+            let router = match &flags.router_mode {
                 RouterMode::Random | RouterMode::RoundRobin => {
-                    client.set_router_mode(flags.router_mode.into());
                     tracing::info!("Waiting for remote model..");
-                    client.wait_for_endpoints().await?;
-                    tracing::info!("Model discovered");
+
+                    // We then use it's `requires_preprocessing` field to decide what kind of
```

Comment:
`it's` is referring to Model Deployment Card here?
```suggestion
                    // We then use the ModelDeploymentCard's `requires_preprocessing` field to decide what kind of
```

Reply:
Done.